timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -200,6 +201,106 @@ void declare_context(py::module &m) {
+                }
+                if (*std::max_element(vec_local_idxs.begin(), vec_local_idxs.end()) >= N) {
+                    throw std::runtime_error("indices values must be less than N");
+                }
+                if (*std::min_element(vec_local_idxs.begin(), vec_local_idxs.end()) < 0) {
+                    throw std::runtime_error("indices values must be greater than or equal to 0");
+                }
+
+                // Verify that local idxs are unique
+                unique_idxs<int>(vec_local_idxs);
+                std::array<std::vector<double>, 2> result =
+                    ctxt.multiple_steps_local(n_steps, vec_local_idxs, burn_in, x_interval, radius, k, seed);
+                const int D = 3;
+                const int F = result[0].size() / (N * D);
+                py::array_t<double, py::array::c_style> out_x_buffer({F, N, D});
+                std::memcpy(out_x_buffer.mutable_data(), result[0].data(), result[0].size() * sizeof(double));
+
+                py::array_t<double, py::array::c_style> box_buffer({F, D, D});
+                std::memcpy(box_buffer.mutable_data(), result[1].data(), result[1].size() * sizeof(double));
+                return py::make_tuple(out_x_buffer, box_buffer);
+            },
+            py::arg("n_steps"),
+            py::arg("local_idxs"),
+            py::arg("burn_in") = 500, // This is arbitrarily selected as a default, TODO make informed choice
+            py::arg("store_x_interval") = 0,
+            py::arg("radius") = 1.2,
+            py::arg("k") = 10000.0,
+            py::arg("seed") = 2022,
+            R"pbdoc(
+        Take multiple steps using particles selected based on the log probability using a random particle from the local_idxs,
+        the random particle is frozen for all steps.
+
+        Running a barostat and local MD at the same time are not currently supported. If a barostat is
+        assigned to the context, the barostat won't run.
+
+        Note: Running this multiple times with small number of steps (< 100) may result in a vacuum around the local idxs due to
+        discretization error caused by switching on the restraint after a particle has moved beyond the radius.
+
+        F = iterations / store_x_interval
+
+        Parameters
+        ----------
+        n_steps: int
+            Number of steps to run.
+
+        local_idxs: np.array of int32
+            The idxs that defines the atoms to use as the region(s) to run local MD. A random idx will be
+            selected to be frozen and used as the center of the shell of particles to be simulated. The selected
+            idx is constant across all steps.
+
+        burn_in: int
+            How many steps to run prior to storing frames. This is to handle the fact that the local simulation applies a
+            restraint, and burn in helps equilibrate the local simulation. Running with small numbers of steps (< 10) is not recommended.
```

Comment:
this is `<10` but above its `<100` - maybe we should be consistent?

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/915/commits/829a8df6e428ded5ebceafd4dcb8da1f7502b49b